File: game_server/ServerProtocol/src/GC2GS/p036_TreasureHuntOp/GC2GS_036_002_ReqTreasureHuntOreSkillActive.java
package GC2GS.p036_TreasureHuntOp;

import java.nio.ByteBuffer;
/*********
 * 太空寻宝-矿石技能激活
 **/
public class GC2GS_036_002_ReqTreasureHuntOreSkillActive implements ALBasicProtocolPack._IALProtocolStructure {
/** 矿石ID */
private long oreId;
/** true普通技能 false高级技能 */
private boolean isNormal;


public GC2GS_036_002_ReqTreasureHuntOreSkillActive() {
	oreId = (long)0;
	isNormal = false;
}

public GC2GS_036_002_ReqTreasureHuntOreSkillActive(
	 long _oreId
	, boolean _isNormal
) {	oreId = _oreId;
	isNormal = _isNormal;
}

public final byte getMainOrder() { return (byte)36; }

public final byte getSubOrder() { return (byte)2; }

/** 矿石ID */
public long getOreId() { return oreId; }
/** 矿石ID */
public void setOreId(long _oreId) { oreId = _oreId; }
/** true普通技能 false高级技能 */
public boolean getIsNormal() { return isNormal; }
/** true普通技能 false高级技能 */
public void setIsNormal(boolean _isNormal) { isNormal = _isNormal; }


public final int GetBufSize() {
	int _size = 9;

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 11;

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) oreId = _buf.getLong();
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) isNormal = (_buf.get() != 0);
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.putLong(oreId);
	_buf.put(isNormal?(byte)1:(byte)0);
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)36);
	_buf.put((byte)2);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)36);
	_recBuf.put((byte)2);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

File: game_server/ClientProtocol/GC2GS/p036_TreasureHuntOp/GC2GS_036_002_ReqTreasureHuntOreSkillActive.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace GC2GS.p036_TreasureHuntOp
{

/// <summary>
/// 太空寻宝-矿石技能激活
/// </summary>
public class GC2GS_036_002_ReqTreasureHuntOreSkillActive : ALBasicProtocolPack._IALProtocolStructure {
/// <summary>
/// 矿石ID
/// </summary>
private long oreId;
/// <summary>
/// true普通技能 false高级技能
/// </summary>
private bool isNormal;


public GC2GS_036_002_ReqTreasureHuntOreSkillActive() {
	oreId = (long)0;
	isNormal = false;
}

public GC2GS_036_002_ReqTreasureHuntOreSkillActive(
	long _oreId
	, bool _isNormal
) {	oreId = _oreId;
	isNormal = _isNormal;
}

public byte getMainOrder() { return (byte)36; }

public byte getSubOrder() { return (byte)2; }

/// <summary>
/// 矿石ID
/// </summary>
public long getOreId() { return oreId; }
/// <summary>
/// 矿石ID
/// </summary>
public void setOreId(long _oreId) { oreId = _oreId; }
/// <summary>
/// true普通技能 false高级技能
/// </summary>
public bool getIsNormal() { return isNormal; }
/// <summary>
/// true普通技能 false高级技能
/// </summary>
public void setIsNormal(bool _isNormal) { isNormal = _isNormal; }


public int GetBufSize() {
	int _size = 9;

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 11;

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	oreId = _buf.getLong();
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	isNormal = (_buf.get() != 0);
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putLong(oreId);
	_buf.put(isNormal?(byte)1:(byte)0);
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)36);
	_buf.put((byte)2);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)36);
	_recBuf.put((byte)2);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("oreId").Append(":").Append(oreId.ToString()).Append(", ");
	builder.Append("isNormal").Append(":").Append(isNormal.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

